ui/button.js

```diff
@@ -56,355 +39,267 @@ $.widget( "ui.button", {
+			if ( key === "showLabel" ) {
+				this.options.text = value;
 			}
-
-			if ( icons.primary ) {
-				buttonElement.prepend( "<span class='ui-button-icon-primary ui-icon " + icons.primary + "'></span>" );
+			if ( key === "icon" ) {
+				this.options.icons.primary = value;
 			}
-
-			if ( icons.secondary ) {
-				buttonElement.append( "<span class='ui-button-icon-secondary ui-icon " + icons.secondary + "'></span>" );
+			if ( key === "icons" ) {
+				if ( value.primary ) {
+					this._setOption( "icon", value.primary );
+					this._setOption( "iconPosition", "beginning" );
+				} else if ( value.secondary ) {
+					this._setOption( "icon", value.secondary );
+					this._setOption( "iconPosition", "end" );
+				}
 			}
-
-			if ( !this.options.text ) {
-				buttonClasses.push( multipleIcons ? "ui-button-icons-only" : "ui-button-icon-only" );
-
-				if ( !this.hasTitle ) {
-					buttonElement.attr( "title", $.trim( buttonText ) );
+			this._superApply( arguments );
+		}
+	});
+	$.fn.button = (function( orig ) {
+		return function() {
+			if ( this.length > 0 && this[ 0 ].tagName === "INPUT" &&
+					( this.attr( "type") === "checkbox" || this.attr( "type" ) === "radio" ) ) {
+				if ( $.ui.checkboxradio ) {
+					if ( arguments.length === 0 ) {
+						return this.checkboxradio({
+							"icon": false
+						});
+					} else {
+						return this.checkboxradio.apply( this, arguments );
+					}
+				} else {
+					$.error( "Checkboxradio widget missing" );
 				}
+			} else {
+				return orig.apply( this, arguments );
 			}
+		};
+	})( $.fn.button );
+	$.fn.buttonset = function() {
+		if ( $.ui.controlgroup ) {
+			if ( arguments[ 0 ] === "option" && arguments[ 1 ] === "items" && arguments[ 2 ] ) {
+				return this.controlgroup.apply( this,
+					[ arguments[ 0 ], "items.button", arguments[ 2 ] ] );
+			} else if ( typeof arguments[ 0 ] === "object" && arguments[ 0 ].items ) {
+				arguments[ 0 ].items = {
+					button: arguments[ 0 ].items
+				};
+			} else if ( arguments[ 0 ] === "option" && arguments[ 1 ] === "items" ) {
+				return this.controlgroup.apply( this, [ arguments[ 0 ], "items.button" ] );
+			}
+			return this.controlgroup.apply( this, arguments );
 		} else {
-			buttonClasses.push( "ui-button-text-only" );
+			$.error( "Controlgroup widget missing" );
 		}
-		buttonElement.addClass( buttonClasses.join( " " ) );
-	}
-});
-
-$.widget( "ui.buttonset", {
-	version: "@VERSION",
-	options: {
-		items: "button, input[type=button], input[type=submit], input[type=reset], input[type=checkbox], input[type=radio], a, :data(ui-button)"
-	},
-
-	_create: function() {
-		this.element.addClass( "ui-buttonset" );
-	},
-
-	_init: function() {
-		this.refresh();
-	},
-
-	_setOption: function( key, value ) {
-		if ( key === "disabled" ) {
-			this.buttons.button( "option", key, value );
-		}
-
-		this._super( key, value );
-	},
-
-	refresh: function() {
-		var rtl = this.element.css( "direction" ) === "rtl",
```

Comment:
Looks like the RTL support from buttonset didn't make it to controlgroup. Is that intended? Since we have plans to look into RTL support for all widgets, it seems fine to me.